File: get.php
<?php

//beginning of endpoint. change if needed
$gate = 'http://gateway.marvel.com/v1/public/';

//Parameters change as you see fit. I picked characters and Spidey
$param = "characters?name=Spider-Man";

//replace with your key from developers.marvel
$APIKey ='YOURKEY123';

//replace with your private key from developers.marvel
$PrivateKey = 'YOURPRIVATEKEY123';

// just used an integer for this example
$timestamp = '1';

//create hash
$hash = md5($timestamp . $PrivateKey . $APIKey);

//build endpoint
$ch = curl_init($gate . $param . "&ts=" . $timestamp . "&apikey=" . $APIKey . "&hash=" . $hash);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_HEADER, 0);
curl_setopt($ch, CURLOPT_HEADER, true );


$data = curl_exec($ch);

$httpcode = curl_getinfo($ch, CURLINFO_HTTP_CODE);

curl_close($ch);

//dump the reposnse. This should look like the interactive docs at developers.marvel
    var_dump(
 $data,
 json_encode($data)
);

?>

File: get.pl
use LWP::Simple;
use Digest::MD5 qw(md5 md5_hex md5_base64);

my $gate = 'http://gateway.marvel.com/v1/public/';
#Parameters change as you see fit. I picked characters and Spidey
my $param = "characters?name=Spider-Man";
#replace with your key from developers.marvel
my $apikey ='YOURKEY123';
#replace with your private key from developers.marvel
my $privatekey = 'YOURPRIVATEKEY123';
# just used an integer for this example
my $timestamp = '1';

#parts of the urls string
my $ts_amp = '&ts=';
my $api_amp = '&apikey=';
my $hash_amp = '&hash=';
my $m = md5_hex($timestamp . $privatekey . $apikey);

my $url = $gate . $param . $ts_amp . $timestamp . $api_amp . $apikey . $hash_amp . $m;
          # ACME boomerang
my $response = get $url;
die 'Error getting $url' unless defined $response;

#print out the response
print $response;
